src/components/views/read/cvForm/Education.tsx

```diff
@@ -166,151 +180,168 @@ const Education: FC<Props> = ({
-                      label="Till"
-                      value={
-                        inputValues.find(
-                          (inputValue) => inputValue.cardId === card.id,
-                        )?.values.endDate || endDate
-                      }
-                      minDate={startDate}
-                      maxDate={dayjs()}
-                      onChange={(date) =>
-                        handleChange(
-                          card.id,
-                          'endDate',
-                          date
-                            ? dayjs(date).format('YYYY-MM-DD')
-                            : dayjs(endDate).format('YYYY-MM-DD'),
-                        )
-                      }
-                    />
-                  </LocalizationProvider>
-                  <TextField
-                    label="GPA"
-                    value={
-                      inputValues.find(
-                        (inputValue) => inputValue.cardId === card.id,
-                      )?.values.gpa || ''
-                    }
-                    onChange={(e) =>
-                      handleChange(card.id, 'gpa', e.target.value)
-                    }
-                  />
-                  <TextField
-                    id={`select-country-${card.id}`}
-                    select
-                    label="Country"
-                    value={
-                      inputValues.find(
-                        (inputValue) => inputValue.cardId === card.id,
-                      )?.values.country || ''
-                    }
-                    onChange={(e) =>
-                      handleChange(card.id, 'country', e.target.value)
-                    }
-                    required
-                    helperText="Please select your country"
-                    margin="normal"
-                  >
-                    <MenuItem value="Switzerland">Switzerland</MenuItem>
-                    <MenuItem value="Europe">Europe</MenuItem>
-                    <MenuItem value="Asia">Asia</MenuItem>
-                    <MenuItem value="Africa">Africa</MenuItem>
-                    <MenuItem value="North America">North America</MenuItem>
-                    <MenuItem value="South America">South America</MenuItem>
-                    <MenuItem value="Other">Other</MenuItem>
-                  </TextField>
+                  {mapping.map((m) => (
+                    <>
+                      <p>{m.label}</p>
+                      {m.key !== 'degree' &&
+                      m.key !== 'startDate' &&
+                      m.key !== 'endDate' &&
+                      m.key !== 'country' ? (
+                        <TextField
+                          label={m.label}
+                          id={m.key}
+                          value={
+                            inputValues.find(
+                              (inputValue) => inputValue.cardId === card.id,
+                            )?.values[m.key] || ''
```

Comment:
Your data structure is too convoluted if you have to do this kind of processing to simply access the value of a field.

It should be as simple as `card[m.key]`. I don't understand why you are maintaining 2 mappings (`cards` and `inputValues` that essentially replicate themselves with different structure...